columnflow/tasks/framework/base.py

```diff
@@ -440,25 +440,43 @@ def get_array_function_kwargs(cls, task=None, **params):
     @classmethod
     def resolve_config_default(
         cls,
-        config_inst: od.Config,
+        task_params: dict[str, Any],
         param: str,
+        container: str | od.AuxDataMixin = "config_inst",
         default_str: str | None = None,
     ) -> str | None:
         """
         Applies the default value identified by *default_str* in the *config_inst*,
         if no *param* is given.
         """
+
+        # get the container inst
+        if isinstance(container, str):
+            container = task_params.get(container, None)
+
+        if not container:
+            return param
+
         # expand default when param is empty
         if not param or param == law.NO_STR:
-            param = config_inst.x(default_str, None) if default_str else None
+            param = container.x(default_str, None) if default_str else None
+
+            # allow default to be a function, taking task parameters as input
+            if isinstance(param, Callable):
+                param = param(container, **task_params)
+
+        # if multiple params are returned, use the first
+        if param and isinstance(param, (list, tuple)):
+            param = param[0]
```

Comment:
We have cases (CalibratorsMixin, ProducersMixing (both plural)) where we might want to define a list/tuple with more than one default. Do you think it would make sense here to add a flag `multiple=False/True` or alike, to signalize that not just the first value should be returned?

Also, if multiple values are expected (`multiple=True`) and the callable just returns a single value (e.g. because the same function was used for "default_producer" and "default_producers"), one could use this info and return a 1-tuple.